### PR TITLE
Addition of makeshift rope for select uses

### PR DESCRIPTION
**Pull request by DangerNoodle**
Implemented variants of rope made from natural cordage as alternatives to rope for some recipes and constructions, as a continuation of a prior pull request. In this case ropes were chosen instead of directly allowing withered plants or straw simply due to the possible abnormality of weaving much larger amounts of material on the spot, and for larger projects allows long makeshift ropes (or multiple short ones) to be made from a mixture of withered plants and piles of straw, instead of always demanding one hundred percent of one option or another, which is useful given the high material demand.

However, I did decide to reduce the material demands of makeshift ropes due to balance concerns, as these will have a very high consumption of withered plants and piles of straw.

Makeshift doors, palisade pulleys, and pontoon bridges were set to return makeshift rope and/or withered plants when taken apart, and have makeshift ropes added to their construction recipes. Transmutation of thread into withered plants is unfortunate but not exploitable, unlike transmutation of withered plants into thread. In the case of palisade pulleys, this additionally fixes the potential (albeit unlikely) exploit of growing vines and converting them into cotton ropes.

Door curtains, being makeshift in nature, have add withered plants and piles of straw added as alternatives to string, and set to return withered plants for the same reasoning as above.

Palisade walls, palisade gates, and construction recipes for mining have also been set to allow makeshift ropes, and these do not return ropes if bashed down, and additionally cannot be deconstructed.

For crafting cargo racks, wooden armor plates, chitin armor plates, bone armor plates, light wooden frames, heavy snare kits have been set to allow makeshift ropes. The former cannot be deconstructed in the first place, and the others have been set to return makeshift ropes by default. In the case of chitin and bone armor plates, this prevents potential transmutation of glue into thread, though glue is situationally more valuable than thread.

Any proposed changes or feedback would be desired.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Needs wrapping in a `requirement`. Preferably with `vine`.

File: data/json/items/generic/string.json [hunk 1]
```diff
@@ -64,5 +64,32 @@
...
     "material": "veggy",
     "color": "light_green"
+  },
+  {
+    "id": "rope_makeshift_6",
+    "copy-from": "rope_6",
+    "type": "GENERIC",
+    "name": "short makeshift rope",
+    "description": "A six-foot long piece of rough rope, woven from natural cordage.  Useful for some purposes, but not as strong or flexible as proper rope.",
+    "material": "paper"
```
Comment (Coolthulhu) [inline data/json/items/generic/string.json]: Wouldn't `veggy` fit better?

**Comment by Coolthulhu** [comment]
Other than that it looks OK.

**Comment by DangerNoodle** [comment]
Will work on requirements, then. As for use of paper, that was picked only due to withered plants and straw counting as paper, can use veggy instead.

**Comment by DangerNoodle** [comment]
Apologies for the delay, should be ready.

File: data/json/requirements/materials.json [hunk 2]
```diff
@@ -45,6 +45,18 @@
     "components": [ [ [ "thread", 1 ], [ "sinew", 1 ], [ "plant_fibre", 1 ], [ "yarn", 1 ] ] ]
   },
   {
+    "id": "rope_natural",
+    "type": "requirement",
+    "//": "Materials used for lashing, when choice of makeshift rope or vine is sensible",
+    "components": [ [ [ "rope_6", 5 ], [ "rope_30", 1 ], [ "rope_makeshift_6", 5 ], [ "rope_makeshift_30", 1 ], ], [ "vine_30", 1 ] ] ]
```
Comment (kevingranade) [inline data/json/requirements/materials.json]: You never loaded this, your json syntax is invalid, even the diff display shows it.

Comment (DangerNoodle) [inline data/json/requirements/materials.json]: That is strange. My apologies for that, I thought that I had copied a prior requirement correectly, but I must have missed it. Do I need to change anything?

**Comment by kevingranade** [comment]
No change needed, I fixed it while merging.

Please test your changes before pushing them, even if it's just loading a
game to have it check the json validity.

**Pull request merged**